src/long.ts

```diff
@@ -263,12 +294,20 @@ export class Long extends BSONValue {
       unsigned = !!unsigned;
     }
     radix = radix || 10;
+
     if (radix < 2 || 36 < radix) throw new BSONError('radix');
 
+    if (throwsError && !Long.validateStringCharacters(str, radix)) {
+      throw new BSONError(`Input: '${str}' contains invalid characters for radix: ${radix}`);
+    }
+    if (throwsError && str.trim() !== str) {
+      throw new BSONError(`Input: '${str}' contains whitespace.`);
```

Comment:
Can we clarify that it's specifically leading or trailing whitespace?  internal whitespace will be caught in the check above